WordPress/src/androidTest/java/org/wordpress/android/e2e/flows/SignupFlow.java

```diff
@@ -0,0 +1,71 @@
+package org.wordpress.android.e2e.flows;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.rule.ActivityTestRule;
+
+import org.wordpress.android.R;
+import org.wordpress.android.ui.accounts.LoginMagicLinkInterceptActivity;
+
+import static android.support.test.InstrumentationRegistry.getTargetContext;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.matcher.ViewMatchers.withHint;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.wordpress.android.support.WPSupportUtils.checkViewHasText;
+import static org.wordpress.android.support.WPSupportUtils.clickOn;
+import static org.wordpress.android.support.WPSupportUtils.populateTextField;
+import static org.wordpress.android.support.WPSupportUtils.waitForElementToBeDisplayed;
+
+public class SignupFlow {
+    public void chooseSignupWithEmail() {
+        clickOn(onView(withId(R.id.create_site_button)));
+        clickOn(onView(withId(R.id.signup_email)));
+    }
+
+    public void enterEmail(String email,
+                           ActivityTestRule<LoginMagicLinkInterceptActivity> magicLinkActivityTestRule) {
+        // Email file = id/input
+        populateTextField(onView(withId(R.id.input)), email);
+        clickOn(onView(withId(R.id.primary_button)));
+
+        // Follow the magic link to continue login
+        // Intent is invoked directly rather than through a browser as WireMock is unavailable once in the background
+        Intent intent = new Intent(
+                Intent.ACTION_VIEW,
+                Uri.parse("wordpress://magic-login?token=valid_token&new_user=1")
+        ).setPackage(getTargetContext().getPackageName());
+
+        magicLinkActivityTestRule.launchActivity(intent);
+    }
+
+    public void checkEpilogue(String email, String displayName, String username) {
+        // Check Epilogue data
+        ViewInteraction emailHeaderView = onView(withId(R.id.signup_epilogue_header_email));
+        waitForElementToBeDisplayed(emailHeaderView);
+        checkViewHasText(emailHeaderView, email);
+
+        ViewInteraction displayNameField = onView(allOf(withId(R.id.input), withText(displayName)));
+        ViewInteraction usernameField = onView(allOf(withId(R.id.input), withText(displayName)));
+
+        waitForElementToBeDisplayed(displayNameField);
```

Comment:
The test is failing on this line for me.

Here is the error:

```
junit.framework.AssertionFailedError: Unable to continue – expectation wasn't satisfied quickly enough
at junit.framework.Assert.fail(Assert.java:50)
at org.wordpress.android.support.WPIdler.idleUntilReady(WPIdler.java:48)
at org.wordpress.android.support.WPIdler.idleUntilReady(WPIdler.java:37)
at org.wordpress.android.support.WPSupportUtils.waitForConditionToBeTrue(WPSupportUtils.java:281)
at org.wordpress.android.support.WPSupportUtils.waitForElementToBeDisplayed(WPSupportUtils.java:236)
at org.wordpress.android.e2e.flows.SignupFlow.checkEpilogue(SignupFlow.java:53)
at org.wordpress.android.e2e.SignUpTests.signUpWithEmail(SignUpTests.java:27)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at android.support.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at android.support.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:527)
at android.support.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:527)
at com.github.tomakehurst.wiremock.junit.WireMockRule$1.evaluate(WireMockRule.java:73)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at android.support.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at android.support.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:384)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
```

And a screenshot where it fails:

![Screenshot_1559223815](https://user-images.githubusercontent.com/1773641/58636970-5f566880-82e9-11e9-8d09-6020a142d23f.png)